tests/test_connection.py

```diff
@@ -129,8 +136,10 @@ async def wrapper(*args, **kwargs):
 @attr.s(hash=False, eq=False)
 class MemoryListener(trio.abc.Listener):
     closed = attr.ib(default=False)
-    accepted_streams = attr.ib(factory=list)
-    queued_streams = attr.ib(factory=lambda: trio.open_memory_channel(1))
+    accepted_streams: list[
+        tuple[trio.abc.SendChannel[str], trio.abc.ReceiveChannel[str]]
+    ] = attr.ib(factory=list)
+    queued_streams = attr.ib(factory=lambda: trio.open_memory_channel[str](1))
```

Comment:
Is this a mistake?  I don't know why the lambda return expression includes a type annotation (`[str]`).

Reply:
the annotation is for `trio.open_memory_channel`, specifying the type of the objects being sent over the channels. It's not normally possible to specify the type on functions, so Trio says it's a callable class during type checking to allow for that usage: https://github.com/python-trio/trio/blob/4dcdc6dbe899c9ee295fa9c8e6a077100ae19f16/src/trio/_channel.py#L91

Without the annotation, `queued_streams` would need the same long & verbose type annotation as `accepted_streams`.